src/views/AuthoritiesSearch/AuthoritiesSearch.js

```diff
@@ -373,13 +377,20 @@ const AuthoritiesSearch = ({
     });
 
     return `${match.path}/authorities/${authority.id}?${newSearch}`;
-  };
+  }, [match.path, location.search]);
 
-  const redirectToAuthorityRecord = authority => {
+  const redirectToAuthorityRecord = useCallback(authority => {
     history.push(formatAuthorityRecordLink(authority));
-  };
+  }, [history, formatAuthorityRecordLink]);
+
+  useEffect(() => {
+    if (!isLoading) {
```

Comment:
I think it's better to fix this logic inside `useAutoOpenDetailView` hook - it already contains info about authorities
So we can change the `useEffect` to only open the detail view when first authority in `authorities` changes and is equal to `selectedAuthority` context

Reply:
Thanks for the comment. `selectedAuthority` becomes `null` after clicking close (X) on the Detail View pane, and the condition `selectedAuthority?.id === firstAuthority.id` will not prevent opening the pane again.







https://user-images.githubusercontent.com/77053927/203541109-04acc390-bc60-4f76-b050-e505bbd1b28c.mp4